packages/common/src/ProviderUtils.ts

```diff
@@ -8,6 +8,11 @@ import minimist from "minimist";
 import Url from "url";
 import { RetryProvider, RetryConfig } from "./RetryProvider";
 import { AbstractProvider } from "web3-core";
+import HDWalletProvider from "@truffle/hdwallet-provider";
```

Comment:
Does this tie us more to `truffle`? Is there a way to use hardhat?

Reply:
Great point!

- I'm not aware of any alternatives provided by hardhat or otherwise, especially that support web3 and not just Ethers.
- This is mainly just a simple web3 provider that just happens to be part of the truffle monorepo. It shouldn't add any reliance to the core of truffle or to using truffle tools since it is immediately plugged into web3.

If you have an equivalent alternative, I'm happy to use it! I think hardhat deploy has an hd wallet config that it plugs into an internal hd wallet implementation, but I don't think that implementation is exported on its own so that you can use it outside the hardhat context (directly with web3, for instance).

Reply:
Yeah, while I was digging through the `hardhat` docs I couldn't find an equivalent for HDWalletProvider and I agree that it doesn't expand the truffle dependence anyways.